app/presenters/finder_presenter.rb

```diff
@@ -64,8 +76,7 @@ def related
 
   def results
     @results ||= ResultSet.get(
-      slug,
-      document_type,
+      self,
```

Comment:
I'm really unsure about this. It makes the owning object unclear.

My gut says to start by removing calls to the `some_finder_presenter.results` method here with direct uses of `ResultSet`.